app/services/parse/google_parse_service.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+module Parse
+  class GoogleParseService
+    SELECTORS = {
+      adwords_top: 'div.pla-unit-container a.pla-unit-title-link',
+      adwords_body: 'div[data-text-ad] div.v5yQqb',
+      non_adwords: 'div#search div.yuRUbf a[data-ved]',
+      href: 'href'
+    }.freeze
+
+    def initialize(html_code)
+      raise IcRailsHueyToby::Errors::ParseServiceError unless html_code
+
+      @html_code = html_code
+    end
+
+    def parse
+      doc = Nokogiri::HTML(html_code)
+
+      adwords_result = parse_adword_list(doc)
+      non_adwords_result = parse_non_adword_list(doc)
+      total_links_count = parse_total_links_count(adwords_result, non_adwords_result)
+
+      adwords_result.merge(non_adwords_result, total_links_count)
```

Comment:
It is a not odd at first as I was looking where was the first hash 😅 How about:

```ruby
Hash.new.merge(adwords_result, non_adwords_result, total_links_count)
```

Reply:
Oh. It looks clearer that we gonna merge 3 hashes.

Reply:
Fixed [b5004c9](https://github.com/nimblehq/ic-rails-huey-toby/pull/37/commits/b5004c9c50c78d795cdabff28210dd26404e18b7)